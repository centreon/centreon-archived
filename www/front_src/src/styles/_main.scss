@import "partials/typography";
@import "partials/fonts";
@import "partials/variables";
@import "partials/messages";
@import "partials/icons";
@import "partials/form/form";
@import "partials/helpers";
@import "partials/mixins";
