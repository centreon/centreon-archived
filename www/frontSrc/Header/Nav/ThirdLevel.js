import React from 'react'
import { withStyles } from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'

const styles = theme => ({
  wrapper: {
    display: 'inline-flex'
  },
  thirdLevel: {
    display: 'flex',
    alignItems: 'flex-start',
    justifyContent: 'flex-start',
  },
  thirdLevelContent: {
    padding: 14,
  },
  paper: {
    padding: theme.spacing.unit,
  },
  popover: {
    pointerEvents: 'none',
  },
  popperClose: {
    pointerEvents: 'none',
  },
  typoGroup: {
    color: '#818285'
  },
  typoItem: {
    color: '#231F20',
    paddingLeft: 6,
    fontSize: 13
  },
})

const ThirdLevel = (
  {classes,
    key,
    thirdLevelArray,})  => (

  <div className={classes.thirdLevel} key={key} >
    { thirdLevelArray.map((item, i) => {
      if (item.hasOwnProperty('children')) {

        return (
          <div  className={classes.thirdLevelContent}>
            <Typography key={i} variant="subheading" className={classes.typoGroup} gutterBottom>{item.label}</Typography>
            {
              item.children.map((item2, i2) => {
                return <Typography key={i2} className={classes.typoItem} gutterBottom>{item2.label}</Typography>
              })
            }
          </div>
      )
      } else return (
          <Typography key={i} className={classes.typoItem} >{item.label}</Typography>
      )
    }
    )}
  </div>
)
export default withStyles(styles)(ThirdLevel)