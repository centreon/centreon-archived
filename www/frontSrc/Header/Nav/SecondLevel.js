import React, { Component } from 'react'
import { withStyles } from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'
import ThirdLevel from './ThirdLevel'
import { Tooltip } from 'reactstrap'

const styles = theme => ({
  wrapper: {
    display: 'inline-flex'
  },
  paper: {
    padding: theme.spacing.unit,
  },
  popover: {
    pointerEvents: 'none',
  },
  popperClose: {
    pointerEvents: 'none',
  },
  navTypo: {
    color: '#FFF',
    padding: '0 14px',
    cursor: 'pointer',
    margin: 0
  },
  tooltip: {
    zIndex: 10,
    backgroundColor: '#FFF',
    padding: 10,
    borderRadius: 2,
    boxShadow: `0px 3px 5px -1px rgba(0, 0, 0, 0.2), 
                0px 6px 10px 0px rgba(0, 0, 0, 0.14), 
                0px 1px 18px 0px rgba(0, 0, 0, 0.12)`,
  }
})

class SecondLevel extends Component {

  constructor(props) {
    super(props)
    this.state = {
      tooltipOpen: false,
    }
  }

  toggle = () => {
    this.setState({
      tooltipOpen: !this.state.tooltipOpen
    })
  }

  render = () => {
    const { tooltipOpen } = this.state
    const { classes, id, item, key} = this.props
    const aria = `tooltip-${id}`

    return (
      <div className={classes.wrapper} id={key}>
        <Typography id={aria} variant="body2" className={classes.navTypo} gutterBottom> {item.label} </Typography>

        {item.children.length > 0 ?
          <Tooltip placement="bottom"
                   isOpen={tooltipOpen}
                   target={aria}
                   autohide={false}
                   toggle={this.toggle}
                   className={classes.tooltip}>
            <ThirdLevel
              thirdLevelArray={item.children}
            />
          </Tooltip>
          : null}
      </div>
    )
  }
}

export default withStyles(styles)(SecondLevel)
