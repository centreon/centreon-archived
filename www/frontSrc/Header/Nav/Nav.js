import React from 'react'
import { withStyles } from '@material-ui/core/styles'
import Tabs from '@material-ui/core/Tabs'
import Tab from '@material-ui/core/Tab'
import HomeIcon from "../Icons/HomeIcon"
import MonitoringIcon from "../Icons/MonitoringIcon"
import ConfigurationIcon from "../Icons/ConfigurationIcon"
import AdministrationIcon from "../Icons/AdministrationIcon"
import ReportingIcon from "../Icons/ReportingIcon"
import SecondLevel from "./SecondLevel"

const styles = theme => ({
  root: {
   /* display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',*/
  },
  tabsRoot: {
    borderBottom: '1px solid #e8e8e8',
  },
  tabsIndicator: {
    backgroundColor: '#1890ff',
  },
  tabRoot: {
    textTransform: 'initial',
    minWidth: 48,
    borderRadius: 48,
    '&:hover': {
      color: '#40a9ff',
      opacity: 1,
    },
    '&$tabSelected': {
      color: '#1890ff',
      fontWeight: theme.typography.fontWeightMedium,
    },
    '&:focus': {
      color: '#40a9ff',
    },
  },
  tabSelected: {},
  wrapper: {
    display: 'inline-flex'
  },
  navIcons: {
  },
  secondLevel: {
    position: 'absolute',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'start',
    left: 0,
    width: '100%',
    height: 34,
    borderTop: '1px solid #fff',
    top: 65,
    boxShadow: `0px 3px 5px -1px rgba(0, 0, 0, 0.2), 
                0px 6px 10px 0px rgba(0, 0, 0, 0.14), 
                0px 1px 18px 0px rgba(0, 0, 0, 0.12)`,
    zIndex: 1,
  },
  typography: {
    padding: theme.spacing.unit * 3,
  },
})

const Nav = (
  {classes, items, value, handleChange, handlePopoverClose, handlePopoverOpen, open
  }) => (
      <div className={classes.root}>
        <Tabs
          value={value}
          onChange={handleChange}
          classes={{ root: classes.tabsRoot, indicator: classes.tabsIndicator }}
        >
          <Tab disableRipple
            icon={<HomeIcon viewBox="6 156 600 600" className={classes.navIcons}/>}
            classes={{ root: classes.tabRoot, selected: classes.tabSelected }}
          />
          <Tab
            icon={<MonitoringIcon viewBox="0 0 600 600"/>}
            classes={{ root: classes.tabRoot, selected: classes.tabSelected }}
          />
          <Tab
            icon={<ReportingIcon viewBox="0 0 600 600"/>}
            classes={{ root: classes.tabRoot, selected: classes.tabSelected }}
          />
          <Tab
            icon={<ConfigurationIcon viewBox="0 0 600 600"/>}
            classes={{ root: classes.tabRoot, selected: classes.tabSelected }}
          />
          <Tab
            icon={<AdministrationIcon viewBox="6 156 600 600"/>}
            classes={{ root: classes.tabRoot, selected: classes.tabSelected }}
          />
        </Tabs>
        {
          items.map((item, index) => {
            if(value === index) {
              return (
                <div className={classes.wrapper} key={index}>
                  <div className={classes.secondLevel} key={index} style={{ backgroundColor: item.color }}>
                    {
                      item.children.map((secondItem, i) => {
                        return <SecondLevel
                        key={i}
                        id={i}
                        item={secondItem}
                        open={open}
                        />
                      })
                    }
                  </div>
                </div>
              )
            }
          })
        }
      </div>
)

export default withStyles(styles)(Nav)
