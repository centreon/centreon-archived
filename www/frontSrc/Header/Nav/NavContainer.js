import React, { Component } from 'react'
import Nav from './Nav'
import {connect} from "react-redux"
import {getNavItems} from "../../webservices/navApi"

class NavContainer extends Component {

  constructor(props) {
    super(props)
    this.state = {
      tooltipOpen: false,
      value: 0,
    }
  }
  componentDidMount = () =>  {
    this.props.getNavItems()
  }

  handleChange = (event, value) => {
    this.setState({ value })
  }

  toggle = () => {
    this.setState({
      tooltipOpen: !this.state.tooltipOpen
    })
  }

  render = () => {
    const {  value, tooltipOpen } = this.state
    const { data, dataFetched } = this.props.nav

    if (dataFetched) {
      const key = data.reduce((acc, item) => {
        if (item.active) acc = item.key
        return acc
      },0)
      const activeItemKey = value !== key ? key : value
        return <Nav
          value={value}
          items={data}
          key={activeItemKey}
          handleChange={this.handleChange}
          toggle={this.toggle}
          tooltipOpen={tooltipOpen}
          open={open}
        />
    } else return null
  }
}

const mapStateToProps = (store) => {
  return {
    nav: store.nav,
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    getNavItems: () => {
      return dispatch(getNavItems())
    },
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(NavContainer)