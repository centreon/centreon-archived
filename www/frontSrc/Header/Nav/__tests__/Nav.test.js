/* eslint-env jest */
import React from 'react'
import Enzyme from 'enzyme'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import Nav from '../NavContainer'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles'

Enzyme.configure({ adapter: new Adapter() })

const theme = createMuiTheme({
  font: {
    openSans: "'Open Sans', Arial, Tahoma, Helvetica, Sans-Serif"
  },
})

describe('Nav', () => {
  const middlewares = [thunk]
  const mockStore = configureStore(middlewares)

  const store = mockStore({
    nav: {
      data: [
        {
          label: "home",
          url: "",
          children: [
            {
              label: "home",
              url: "",
              children: []
            }
          ],
        }
      ],
    }
  })

  let getNavItems = jest.fn()

  const component = mount(
    <MuiThemeProvider theme={theme}>
      <Nav store={store} getUser={getNavItems} />
    </MuiThemeProvider>
  )

  it('Nav Component renders correctly', () => {
    const NavComponent = shallow(<Nav store={store} />)
    expect(toJson(NavComponent)).toMatchSnapshot()
  })
})