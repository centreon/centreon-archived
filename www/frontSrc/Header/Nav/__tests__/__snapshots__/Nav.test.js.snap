// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Nav Component renders correctly 1`] = `
<NavContainer
  getNavItems={[Function]}
  nav={
    Object {
      "data": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "label": "home",
              "url": "",
            },
          ],
          "label": "home",
          "url": "",
        },
      ],
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
